File: app/src/main/java/com/productions/gizzmoo/pokemonpuzzleleague/GameDialogFragment.java
package com.productions.gizzmoo.pokemonpuzzleleague;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.app.DialogFragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

public class GameDialogFragment extends DialogFragment {

    private static final String ARG_PARAM1 = "param1";
    private boolean mDidWin;
    private OnGameEndingDialogFragmentReturnListener mListener;

    public GameDialogFragment() {}

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        if (mDidWin) {
            builder.setMessage(R.string.win);
        } else {
            builder.setMessage(R.string.lose);
        }

        builder.setPositiveButton(R.string.okay, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                if (mListener != null) {
                    mListener.onGameEndingDialogResponse();
                }
            }
        });
        return builder.create();
    }

    public static GameDialogFragment newInstance(boolean didWin) {
        GameDialogFragment fragment = new GameDialogFragment();
        Bundle args = new Bundle();
        args.putBoolean(ARG_PARAM1, didWin);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mDidWin = getArguments().getBoolean(ARG_PARAM1);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_game_ending_dialog, container, false);

        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnGameEndingDialogFragmentReturnListener) {
            mListener = (OnGameEndingDialogFragmentReturnListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnGameEndingDialogFragmentReturnListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener.onGameEndingDialogResponse();
        mListener = null;
    }


    public interface OnGameEndingDialogFragmentReturnListener {
        void onGameEndingDialogResponse();
    }
}

File: app/src/main/java/com/productions/gizzmoo/pokemonpuzzleleague/puzzlegame/GameDialogFragment.kt
package com.productions.gizzmoo.pokemonpuzzleleague.puzzlegame

import android.app.AlertDialog
import android.app.Dialog
import androidx.fragment.app.DialogFragment
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.productions.gizzmoo.pokemonpuzzleleague.R

class GameDialogFragment : DialogFragment() {
    private var mDidWin: Boolean = false
    private var mListener: GameEndingDialogListener? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mDidWin = arguments?.getBoolean(ARG_PARAM1) ?: false
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
        inflater.inflate(R.layout.game_fragment_ending_dialog, container, false)

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)

        if (mDidWin) {
            builder.setTitle(R.string.win)
        } else {
            builder.setTitle(R.string.lose)
        }

        builder.setPositiveButton(R.string.okay) { _, _ ->
            mListener?.onGameEndingDialogResponse(mDidWin)
        }

        return builder.create()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is GameEndingDialogListener) {
            mListener = context
        } else {
            throw RuntimeException("$context must implement OnGameEndingDialogFragmentReturnListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener?.onGameEndingDialogResponse(mDidWin)
        mListener = null
    }

    companion object {
        private const val ARG_PARAM1 = "param1"
        fun newInstance(didWin: Boolean): GameDialogFragment {
            val fragment = GameDialogFragment()
            val args = Bundle().apply {
                putBoolean(ARG_PARAM1, didWin)
            }
            fragment.arguments = args
            return fragment
        }
    }
}